src/DurableTask.Netherite/Connections/ConnectionInfoExtensions.cs

```diff
@@ -106,98 +105,119 @@ public static Azure.Storage.Blobs.BlobServiceClient GetAzureStorageV12BlobServic
             }
         }
 
-
         /// <summary>
-        /// Creates an Event Hub client for the given connection info.
+        /// Creates an Event Hub Consumer client for the given connection info.
         /// </summary>
         /// <param name="connectionInfo">The connection info.</param>
-        /// <param name="eventHub">The event hub name.</param>
+        /// <param name="eventHubName">The event hub name.</param>
         /// <returns></returns>
-        public static EventHubClient CreateEventHubClient(this ConnectionInfo connectionInfo, string eventHub)
+        public static EventHubConsumerClient CreateEventHubConsumerClient(this ConnectionInfo connectionInfo, string eventHubName)
```

Comment:
This is the methods I added. Create a `EventHubConsumerClient` to get the partitions information of the eventhub.

Reply:
I think the `EventHubConsumerClient` is the wrong type of client for this purpose. There are two clients, a producer client and a consumer client. Technically, the scale monitor is neither producing nor consuming, so it may appear that it does not matter. However, the consumer clients are actually "heavy" objects that maintain a constant connection to the event hub and for which there is a global limit as to how many may exist. So, we should use the producer client for telemetry. If ound that this is explained here: https://azuresdkdocs.blob.core.windows.net/$web/dotnet/Azure.Messaging.EventHubs/5.9.2/index.html#key-concepts